zebra/zapi_msg.c

```diff
@@ -232,11 +232,6 @@ int zsend_interface_link_params(struct zserv *client, struct interface *ifp)
 {
 	struct stream *s = stream_new(ZEBRA_MAX_PACKET_SIZ);
 
-	if (!ifp->link_params) {
-		stream_free(s);
-		return 0;
```

Comment:
I think that removing only this line is sufficient. As mention, when we disable the link-params, there are not send to the zebra client while it should send an empty link-params to advertise that they have been all deleted.

Reply:
Similar lines 242 and 243 should be remove too